keras_cv/models/backbones/detectron2/detectron2_backbone.py

```diff
@@ -0,0 +1,203 @@
+
+        for i in range(depth):
+            x = WindowedTransformerEncoder(
+                project_dim=embed_dim,
+                mlp_dim=mlp_dim,
+                num_heads=num_heads,
+                use_bias=use_bias,
+                use_rel_pos=use_rel_pos,
+                window_size=window_size
+                if i not in global_attention_indices
+                else 0,
+                input_size=(img_size // patch_size, img_size // patch_size),
+            )(x)
+        x = keras.models.Sequential(
+            [
+                keras.layers.Conv2D(
+                    filters=out_chans, kernel_size=1, use_bias=False
+                ),
+                keras.layers.LayerNormalization(epsilon=1e-6),
+                keras.layers.Conv2D(
+                    filters=out_chans,
+                    kernel_size=3,
+                    padding="same",
+                    use_bias=False,
+                ),
+                keras.layers.LayerNormalization(epsilon=1e-6),
+            ]
+        )(x)
+
+        super().__init__(inputs=img_input, outputs=x, **kwargs)
+
+        self.patch_size = patch_size
+        self.embed_dim = embed_dim
+        self.depth = depth
+        self.mlp_dim = mlp_dim
+        self.num_heads = num_heads
+        self.out_chans = out_chans
+        self.use_bias = use_bias
+        self.use_rel_pos = use_rel_pos
+        self.use_abs_pos = use_abs_pos
+        self.window_size = window_size
+        self.global_attention_indices = global_attention_indices
+        self.layer_norm_epsilon = layer_norm_epsilon
+        self.input_tensor = input_tensor
+        self.include_rescaling = include_rescaling
+
+    @property
+    def pyramid_level_inputs(self):
+        raise NotImplementedError(
+            "The `ViTDetBackbone` model doesn't compute"
+            " pyramid level features."
+        )
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "input_shape": self.input_shape,
+                "input_tensor": self.input_tensor,
+                "include_rescaling": self.include_rescaling,
+                "patch_size": self.patch_size,
+                "embed_dim": self.embed_dim,
+                "depth": self.depth,
+                "mlp_dim": self.mlp_dim,
+                "num_heads": self.num_heads,
+                "out_chans": self.out_chans,
+                "use_bias": self.use_bias,
+                "use_abs_pos": self.use_abs_pos,
+                "use_rel_pos": self.use_rel_pos,
+                "window_size": self.window_size,
+                "global_attention_indices": self.global_attention_indices,
+                "layer_norm_epsilon": self.layer_norm_epsilon,
+            }
+        )
+        return config
+
+    @classproperty
+    def presets(cls):
+        """Dictionary of preset names and configurations."""
+        return copy.deepcopy(backbone_presets)
+
+    # @classproperty
```

Comment:
stray comment?

Reply:
This method loads the presets with weights. I will uncomment it later once the model layers are finalized and the final weights are uploaded.